### PR TITLE
backout: accept multiple revisions to back out

### LINKED ISSUE #3339: FR: `jj backout` should be able to revert an entire series of revisions
It's useful to occasionally not just backout one commit, but a series that landed and ultimately caused regressions. At the moment, `jj backout` takes exactly one revision, so doing this requires scripting around `jj log` in your shell to get the revisions.

Realistically speaking, this means that `jj backout -r` should take a revset, much like `jj abandon` does.

There are some edge cases to handle I think, because you need to effectively resolve the union of all provided revsets, then for any changes that are determined to have a parent/child relationship, you need to back them out in reverse order. But I don't think this would be too hard, so I'm tagging this with "good first issue" for anyone who wants to try it.

### PR DESCRIPTION
**Pull request by bnjmnt4n**
Closes #3339.

# Checklist

If applicable:
- [X] I have updated `CHANGELOG.md`
- [NA] I have updated the documentation (README.md, docs/, demos/)
- [NA] I have updated the config schema (cli/src/config-schema.json)
- [X] I have added tests to cover my changes
-

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 * `jj rebase --skip-empty` has been renamed to `jj rebase --skip-emptied`
 
+* `jj backout --revision` has been renamed to `jj backout --revisions`.
+  The short alias `-r` is still supported.
```
Comment (yuja) [inline CHANGELOG.md:16]: nit: might be better to leave flag name unchanged or aliased to singular form.

(This applies to various commands, and can be addressed later.)

**Review by yuja — APPROVED** [review]
Nice.

File: CHANGELOG.md (hunk 1 above)
Comment (bnjmnt4n) [reply CHANGELOG.md:16]: Right, I changed this for rebase as well. I remember there were at least some other commands which used `--revisions`. It seemed kind of inconsistent for now, which is why I changed it.

Comment (bnjmnt4n) [reply CHANGELOG.md:16]: I will merge this first. Perhaps we can change the flag name later.

**Pull request merged**